test/convection/model/test_validation.rb

```diff
@@ -0,0 +1,184 @@
+require 'test_helper'
+require 'json'
+require 'pp'
+
+class TestValidations < Minitest::Test
+
+  def setup
+    @within_limits = ::Convection.template do
+      description 'Validations Test Template - Within Limits'
+      10.times do |count|
+        resource "EC2_INSTANCE_#{count}" do
+          type 'AWS::EC2::Instance'
+          property 'AvailabilityZone', 'us-east-1a'
+          property 'ImageId', 'ami-76e27e1e'
+          property 'KeyName', 'test'
+          property 'SecurityGroupIds', ['sg-dd733c41', 'sg-dd738df3']
+          property 'Tags', [{
+            'Key' => 'Name',
+            'Value' => 'test-1'
+            }]
+          end
+        end
+      10.times do |count|
+        mapping "Mapping_#{count}" do
+          item 'one', 'two', 'three'
+          item '1', '2', '3'
+        end
+      end
+      10.times do |count|
+        output "Output_#{count}" do
+          description 'An Important Attribute'
+          value get_att('Resource', 'Attribute')
+        end
+      end
+      10.times do |count|
+        parameter "Parameter_#{count}" do
+          type 'String'
+          description 'Example Parameter'
+          default 'm3.medium'
+        end
+      end
+    end
+    @excessive_bytesize = ::Convection.template do
+      description 'Validations Test Template - Excessive Bytesize'
+      200.times do |count|
+        resource "EC2_INSTANCE_#{count}" do
+          type 'AWS::EC2::Instance'
+          property 'AvailabilityZone', 'us-east-1a'
+          property 'ImageId', 'ami-76e27e1e'
+          property 'KeyName', 'test'
+          property 'SecurityGroupIds', ['sg-dd733c41', 'sg-dd738df3']
+          property 'Tags', [{
+            'Key' => 'Name',
+            'Value' => 'test-1'
+            }]
+          end
+        end
+      100.times do |count|
+        mapping "Mapping_#{count}" do
+          item 'us-east-1', 'test', 'cf-test-keys'
+          item 'us-west-1', 'test', 'cf-test-keys'
+        end
+      end
+    end
+    @excessive_description = ::Convection.template do
+      description "0" * 1025
+    end
+    @excessive_resources = ::Convection.template do
+      description 'Validations Test Template - Too Many Resources'
+      201.times do |count|
+        logs_log_group "Log_Group_#{count}" do
+          retention_in_days 365
+        end
+      end
+    end
+    @excessive_resource_name = ::Convection.template do
+      description 'Validations Test Template - Excessive Resource Name'
+        logs_log_group "0"*256  do
```

Comment:
Also introduce an intention revealing variable name instead of having an at-first-glance-random call to `String#*`.